// Copyright 2024 The TensorFlow Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package tensorflow.mlir.tf2xla.internal;

import "tensorflow/core/framework/function.proto";
import "tensorflow/core/framework/tensor_shape.proto";
import "tensorflow/core/protobuf/tpu/compile_metadata.proto";

option cc_enable_arenas = true;

// The input to LegalizeMlirToHlo to reproduce a legalization failure.
message LegalizeMlirToHloReproducer {
  // The TPUCompileMetadataProto used during legalization.
  tensorflow.tpu.TPUCompileMetadataProto compile_metadata = 1;

  // The input shapes used during legalization.
  repeated tensorflow.TensorShapeProto input_shapes = 2;

  oneof computation {
    // The MLIR module to be legalized.
    string mlir_module = 3;
    // The FunctionDefLibrary to be legalized.
    tensorflow.FunctionDefLibrary function_def_library = 4;
  }
}
